sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -1569,6 +1575,49 @@ case class GpuLast(child: Expression, ignoreNulls: Boolean)
       TypeCheckSuccess
     }
   }
+
+  // GENERAL WINDOW FUNCTION
+  override lazy val windowInputProjection: Seq[Expression] = inputProjection
+  override def windowAggregation(
+      inputs: Seq[(ColumnVector, Int)]): RollingAggregationOnColumn =
+    RollingAggregation.nth(-1, if (ignoreNulls) NullPolicy.EXCLUDE else NullPolicy.INCLUDE)
+        .onColumn(inputs.head._2)
+}
+
+case class GpuNthValue(child: Expression, offset: Expression, ignoreNulls: Boolean)
+  extends GpuAggregateWindowFunction
+        with ImplicitCastInputTypes
+        with Serializable {
+
+  // offset is foldable, get value as Spark does
+  private lazy val offsetVal = offset.eval().asInstanceOf[Int]
+
+  // Copied from First
+  override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType, BooleanType)
+  override def children: Seq[Expression] = child :: Nil
+  override def nullable: Boolean = true
+  override def dataType: DataType = child.dataType
+  override def toString: String = s"gpu_nth_value($child, $offset)" +
+      s"${if (ignoreNulls) " ignore nulls"}"
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    val defaultCheck = super.checkInputDataTypes()
+    if (defaultCheck.isFailure) {
+      defaultCheck
+    } else {
+      TypeCheckSuccess
+    }
+  }
+
+  // GENERAL WINDOW FUNCTION
+  override lazy val windowInputProjection: Seq[Expression] =
+    Seq(child, GpuLiteral(ignoreNulls, BooleanType))
```

Comment:
Why is `ignoreNulls` turned into a column? It looks like it is never touched when doing any actual computation so it is just a waste for resources.